app/Service/ProductService.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Service;
+
+use Carbon\Carbon;
+
+class ProductService
+{
+    public const DATE_ADDED_COLUMN = 'date_added';
+    public const PRICE_COLUMN = 'price';
+    public function parseProducts()
+    {
+        $csvFile = storage_path('/app/products.csv');
+        $fileHandle = fopen($csvFile, 'r');
+        while (!feof($fileHandle)) {
+            $lineOfText[] = fgetcsv($fileHandle, 0, ',');
+        }
+        fclose($fileHandle);
+
+        if (empty($lineOfText)) {
+            return [
+                'header' => [],
+                'products' => []
+            ];
+        }
+
+        $header = array_shift($lineOfText);
+
+        $products = collect($lineOfText)->map(function ($item) use ($header) {
+            $result = [];
+            foreach ($header as $index => $value) {
+                if ($value === self::DATE_ADDED_COLUMN) {
+                    $result[$value] = Carbon::parse($item[$index]);
+                    continue;
+                }
+                $result[$value] = $item[$index];
+            }
+
+            return $result;
+        });
```

Comment:
it's a bit unclear what this does at a first glance. I need to run the code in my head to fully understand how $products is structured

For starters, I would move it to another function with a meaningful name and explain how the return value is structured.